kitsune/search/v2/base.py

```diff
@@ -0,0 +1,48 @@
+from django.utils import timezone
+
+from elasticsearch_dsl import Document as DSLDocument
+from elasticsearch_dsl import field
+
+
+class SumoDocument(DSLDocument):
+    """Base class with common methods for all the different documents."""
+
+    indexed_on = field.Date()
+
+    @classmethod
+    def prepare(cls, instance):
+        """Prepare an object given a model instance"""
+        obj = cls()
+        doc_mapping = obj._doc_type.mapping
+        fields = [f for f in doc_mapping]
+        fields.remove("indexed_on")
+
+        # Loop through the fields of each object and set the right values
+        for f in fields:
+
+            try:
+                # This will allow child classes to have own methods in the form of prepare_field
+                prepare_method = getattr(obj, "prepare_{}".format(f))
+                value = prepare_method(instance)
+            except AttributeError:
```

Comment:
I had a few problems before with this `except AttributeError` catching errors within the prepare_method itself, for instance if the method was `return instance.some_attribute_which_doesnt_exist`. Is there some non try/except method for detecting whether a prepare_method exists?